src/components/inventory.rs

```diff
@@ -310,8 +317,8 @@ impl GameEntity<(&mut Context, &mut Player, &mut Inventory, &mut HUD)> for Inven
 
         let (item_cursor_frame, weapon_cursor_frame) = match self.focus {
             InventoryFocus::None => (1, 1),
-            InventoryFocus::Weapons => (1, self.tick & 1),
-            InventoryFocus::Items => (self.tick & 1, 1),
+            InventoryFocus::Weapons => (1, (self.tick / 2) % 2), //every-other frame (& 1): this is not what we want, we want every 2 frames.
```

Comment:
`self.ticks % 2` is indeed identical to `self.ticks & 1`. It will cycle between `1` and `0` every other number.

Example:
`0%2 = 0`, `1%2 = 1`, `2%2 = 0`, `3%2 = 1`

`(self.ticks / 2) % 2` is not. It will cycle between `1` and `0` every *two* numbers, hence the `/2`.

Example:
`0/2 = 0 then 0%2 = 0`, `1/2 = 0 then 0%2 = 0`, `2/2=1 then 1%2=1`, `3/2=1 then 1%2=1`
As you can see, a value of `1` results in `0`, and a value of `2` results in `1`, which is *not* the same as the original code.

If you compared the behavior of the original code with that of the revised code on a 50/60Hz monitor, you will see an obvious difference. The cursor does not appear to blink at all when seen on these screens. *(Additionally, the original game also blinks at this slower, 2-frames-per-cycle speed)*

Reply:
`(self.tick / 2) % 2` can be simplified to `self.tick & 2`.

Reply:
I remember thinking about this when changing the original code, but I couldn't find a concise way to ensure `self.tick & 2` will always be `0` or `1`.

In C you could easily do `!!(self.tick & 2)`
How do you do this in Rust?

The best I could do was `((self.tick & 2) != 0) as usize`, and that seems worse than `(self.tick / 2) % 2`.